app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java

```diff
@@ -1437,6 +1473,7 @@ private BuildStatus getVDBService(String openShiftName, String namespace, final
                     DeploymentCondition cond = getDeploymentConfigStatus(dc);
                     if (cond != null) {
                         status.setStatusMessage(cond.getMessage());
+                        status.setDeploymentVersion(getDeployedRevision(dc, status.getDeploymentVersion(), client));
```

Comment:
@shawkins They are different. I am not assuming one from the other. From the latest build or deployment, I am finding the label for DEPLOYMENT_VERSION_LABEL then using that. "deployment-config.latest-version" or "build.latestversion" are values that Openshift maintains to keep track of what it its latest builds/deployment

Reply:
Okay, I see the differentiation now.  The next comments are:

1. So are we really tracking two things -  The "latest" build (the deploymentVersion from line 1394), and what is currently deployed, but only have one status build object for that?  Is it clear from the status which one of those things we are conveying?
2. In several places == is being used with Long values - that needs converted to something safer.
3. change defaltNumber to defaultNumber

Reply:
@shawkins 
1. No, we are only tracking one thing in this single status call. what the latest version and what is the status of that. As per what is deployed, I have not seen that is being used by UI. I told mark that if he needs that, we can extend the status api with revision number appended to it, then the status of that deployment can be provided. Or we would need to change the current response to a collection.
2 & 3 : sounds good, will have those changes shortly

Reply:
BTW, I do think that we need an API of what is running, but that needs to be tracked in separate JIRA

Reply:
> what the latest version and what is the status of that

I'm asking because we have two.  We're getting:

String deploymentVersion = build.getMetadata().getLabels().get(DEPLOYMENT_VERSION_LABEL);

which I'm calling the deployment version of the latest build, and  

status.setDeploymentVersion(getDeployedRevision(dc, status.getDeploymentVersion(), client));

which is the deployment version from the latest deployment config.  I'm trying to make sure there is clarity on when and why we report one over the other.

Reply:
@shawkins It will not get there if the build has an error. Only build passed and deployment in progress it will report that.

Reply:
@rareddy so the BuildStatus if there is an error will return the deployed version as what was last attempted in the build.  If the last build is successfully BuildStatus will return the deployed version as what was last successfully deployed - with the current versioning/deployment strategy under what circumstances would that be different than the version of the last successful build?

Reply:
@shawkins I am still not getting notified from git for some reason, sigh.

I think the way it worked before it gave a version of deployment that is running, no matter what the build succeeded or not. With this PR, it will only give the status of the revision, from a failed build or successful build. If the build is successful, then there will be a "deployment", and then it gives the status of "that" deployment. Now, this deployment may have failed and rolled to the previous version, but it will still give the revision + status of the failed one, NOT the one that is running. if we want the status of running deployment, then IMO we need to either amend this PR then it will hide the fact that deployment failed, or we can extend the API to provide
`/vitrualization/{name}/revision`
to query individual version, or something like
`/vitrualization/{name}?status=running`

I am not sure how this worked before, so let me know what you want me to change, we can proceed with that.

Reply:
@rareddy

> I am not sure how this worked before, so let me know what you want me to change, we can proceed with that.

It looks like before it just got the revision from the last (the 0 element in the list) build, so it only considered the build - and potentially the wrong build from the looks of your new logic.

It may be cleaner to just have the BuildStatus object remain the last build status - you can even rename the deployed revision field as "buildRevision" and remove some of the deployment specific fields to drive that home more.  Then return a different/new object from getVirtualizationStatus - that would contain both the latest BuildStatus and the latest deployment status (usedBy, deployed revision, routes, deployment status, deployment status message).

If it seems cleaner to have new service methods and/or you want to design for the future and account for different build/deployment states per revision, we can do that as well.

Reply:
This is roughly what I'm talking about: https://github.com/syndesisio/syndesis/compare/master...shawkins:TEIIDTOOLS-914?expand=1

That refines out a separate DeploymentStatus, which can have it's own status, message, and version set.  If we overlay your changes we can track right things individually.

However what I'm seeing is that callers don't really make use of the various fields on the status objects, so I could just simplify this back to a slightly expanded BuildStatus object.